index.js

```diff
@@ -53,6 +53,20 @@ SecurePassword.prototype.hashSync = function (passwordBuf) {
 }
 
 SecurePassword.prototype.hash = function (passwordBuf, cb) {
+  // support promises
+  if (cb === undefined) {
+    return new Promise((resolve, reject) => {
+      this.hash(passwordBuf, function (err, hashBuf) {
+        if (err) {
+          reject(err)
+          return
+        }
+
+        resolve(hashBuf)
+      })
+    })
+  }
```

Comment:
[catering](https://github.com/vweevers/catering) might interest you, wrote it 2 days ago. Would reduce the above to:

```js
const catering = require('catering').attach

SecurePassword.prototype.hash = function (passwordBuf, cb) {
  cb = catering(cb)
  // ..
  return cb.promise
}
```